lib/trivial.nix

```diff
@@ -103,6 +103,9 @@ rec {
   /* bitwise “not” */
   bitNot = builtins.sub (-1);
 
+  /* A variant of the `or` keyword that works with `null` values. */
+  maybe = x: y: if x != null then x else y;
```

Comment:
This isn't Haskell's `maybe`, and it doesn't help readability at all, and this isn't even `Maybe` but "Nullable", which, crucially, can not have a type allowing `null` as its type argument, making it a very leaky abstraction. We really must not think about `null` as `Nothing :: Maybe a`.

Furthermore, your change doesn't need this function. Please remove it.

Reply:
I'm happy to remove it if we can't find an agreement, but I do feel like this is a pattern worth simplifying (if only to avoid duplicating `x`).

How about a more descriptive name like `ifNull x y`?

Reply:
`if x != null then x else y` is about as descriptive as it gets, it's easier to adapt to new requirements and it's faster.

I like combinators as much as you do, but we don't have the luxuries of Haskell here.